enhancements/installer/openstack-availibility-zones.md

```diff
@@ -0,0 +1,125 @@
+---
+title: openstack-availibility-zones
+authors:
+  - "@egarcia"
+  - "@pprenetti"
+reviewers:
+  - "@mandre"
+  - "@fedosin"
+approvers:
+creation-date: 2020-07-15
+last-updated: 2020-07-15
+status: implementable
+---
+
+# OpenStack Availability Zones
+
+## Release Signoff Checklist
+
+- [x]  Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+- Will the node's AZ be used for its volume? Will this effect the behavior of the cluster?
+- Can persistent volume claims be made against volumes that are in a different AZs?
```

Comment:
This is possible if you set the `ignore-volume-az` option to true in the openstack cloud provider configuration, however it is not the default. See https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#block-storage

If needed, modifying the cloud provider config should be done as a day 2 operation.

For us, it means that a deployment on a cloud where cinder AZs do not match nova AZs could potentially hang if swift service isn't available. In which case, CIRO would fallback to cinder for the image-registry and wouldn't be able to attach the volume. This may be resolved by changing the cloud provider config and waiting for the cluster to stabilize with the `openshift-install wait-for install-complete` command. We'll need documentation around it.

We don't want to get involved in the business of setting the `ignore-volume-az` option in the cloud provider config automatically if we detect that the zones don't match. This should instead be a conscious decision by the user. And I don't think we want to expose the cloud provider setting at install time either, at least not right now.